app/controllers/spree/admin/products_controller_decorator.rb

```diff
@@ -78,8 +78,13 @@ def collection
     params[:q][:deleted_at_null] ||= "1"
 
     params[:q][:s] ||= "name asc"
-
-    @search = Spree::Product.ransack(params[:q]) # this line is modified - hit Spree::Product instead of super, avoiding cancan error for fetching records with block permissions via accessible_by
+    # The next line is modified.
+    # Hit Spree::Product instead of super, avoiding cancan error for fetching
+    # records with block permissions via accessible_by.
+    @collection = Spree::Product
+    @collection = @collection.with_deleted if params[:q].delete(:deleted_at_null).blank?
```

Comment:
why is this needed now? AFAIK ransack's [deleted_at_null predicate](https://github.com/activerecord-hackery/ransack/wiki/Basic-Searching#null) should already take the `deleted_at` column into account. From what I understand, if `:deleted_at_null` is 1, it'll fetch records where `deleted_at` is null, while if it's 0 it will fetch them all.

Reply:
This is Spree code from radar's commit https://github.com/openfoodfoundation/spree/commit/cd3add960ef04f5881b508a07a503b3f7cade883#diff-08c8984d7cc18aa24712d5b26ec7defbR88.

The difference in Spree 2 is that `deleted_at is null` is now a default scope. If the ransack parameter `:deleted_at_null` is `true`, it will append the same condition again (not having any effect. And if `:deleted_at_null` is `false`, ransack will not modify the query and it has no effect either. The other solution would be to change the UI to send a `deleted_at_not_null` parameter. But I just went with original Spree code here.